haystack/schema.py

```diff
@@ -1430,3 +1469,36 @@ def load(cls, load_dir: Union[str, Path], **read_csv_kwargs):
             df.rename(columns={"gold_document_contents": "gold_contexts", "content": "context"}, inplace=True)
         result = cls(node_results)
         return result
+
+    @classmethod
+    def load_excel(cls, load_file: Union[str, Path], **read_excel_kwargs):
+        """
+        Loads the evaluation result from disk. Expects an Excel file with one sheet per node. See `save_excel()` for further information.
+
+        :param load_file: Path to the Excel file.
+        :param read_excel_kwargs: The kwargs you want to pass to pd.read_excel(). For more information, see [pandas documentation](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.read_excel.html).
+        """
+        load_file = load_file if isinstance(load_file, Path) else Path(load_file)
+        xlsx = pd.ExcelFile(load_file)
+        default_read_excel_kwargs: Dict[str, Any] = {}
+        read_excel_kwargs = {**default_read_excel_kwargs, **read_excel_kwargs}
+        node_results = {
+            sheet_name: pd.read_excel(load_file, sheet_name, **read_excel_kwargs) for sheet_name in xlsx.sheet_names
+        }
+        # manually restore some column types which were not correctly infered
+        for df in node_results.values():
+            for column in df:
+                if column != "index" and df[column].dtype == "int64":
+                    df[column] = df[column].astype("float64")
+                elif (
+                    df[column].dtype == "object"
+                    and len(df) > 0
+                    and df[column].notna().iloc[0]
+                    and df[column].iloc[0].startswith("[")
+                ):
+                    df[column] = df[column].apply(ast.literal_eval)
+        # backward compatibility mappings
```

Comment:
Maybe we do not need backward compatibility mappings (taken from the original `load()` function), since no Excel file have been saved with previous key names.

Reply:
I deleted this part. Please tell me if you think this could be a problem.